### PR TITLE
Individual parents merge

### LINKED ISSUE #852: Add pedigree data to individual table.
Working on the (not ready for use) pedigree simulation in msprime, I realised that adding pedigree data into the tskit data model would be pretty straightforward and may be of more general use as we get into more detailed datasets. Sometimes we do know who the mother and father of individuals are, and we should be able to record this in the data model. I think the minimum we need to do this is to introduce a new ragged ``parents`` column, which contains the ID(s) of the individual's parents. Maybe we should also add a ``sex`` column, but that's not so clear as it might be as well encoding that in the ``flags`` column. Unless we actually need it, perhaps we're better off leaving that can of worms for now.

I don't think there's any backwards compatibility issues with adding this new column - for old tree sequences, we just regard the ``parents`` list as empty for all individuals.

Should this be data or metadata? I think it's reasonable to assume that we'll have methods in tskit that combine (or at least check the consistency of) information from the pedigree and trees, so it seems like it should be something that we have within tskit's data model.

Beyond enabling msprime's pedigree simulation, I can see this being useful for other types of simulation where perhaps we want to record the individual relationships as well as the genetic. 

Pinging @petrelharp @bhaller @molpopgen @DomNelson @ivan-krukov for thoughts/opinions.

### PR DESCRIPTION
**Pull request by ivan-krukov**
## Description
Adding a ragged `parents` column to the `individuals` table.

Fixes #852 

This is a clean version of the #866 PR

# PR Checklist:

- [x] Tests that fully cover new/changed functionality.
- [ ] Documentation including tutorial content if appropriate.
- [ ] Changelogs, if there are API changes.

### CODE REVIEW

**Comment by benjeffery** [comment]
Nice one on getting this up-to-date! I'll have a detailed look tomorrow.

**Comment by ivan-krukov** [comment]
The final rebase was pretty simple - the patching process took care of everything.
Thank you for writing the detailed how-to `git` guides - they are a treasure.

But it looks like the same test coverage check is failing this time.

There is also an outstanding issue of documenting these changes. Does this need to happen before the merge?

File: c/tests/testlib.c
Comment (jeromekelleher) [inline c/tests/testlib.c:93]: There's still only two and no offspring here, right?

File: docs/metadata.rst
Comment (jeromekelleher) [inline docs/metadata.rst:84]: I think this got zapped in the merge, please revert.

File: python/CHANGELOG.rst
Comment (jeromekelleher) [inline python/CHANGELOG.rst:18]: This is also a merge artefact.

File: python/_tskitmodule.c
Comment (jeromekelleher) [inline python/_tskitmodule.c:352]: Can reuse ``dims`` here for consistency - slightly confusing for the reader to use ``dims`` for location and ``nparents`` for parents.

File: python/tests/test_metadata.py
Comment (jeromekelleher) [inline python/tests/test_metadata.py:1009]: Merge artefact

File: python/tests/test_metadata.py
Comment (jeromekelleher) [inline python/tests/test_metadata.py:1514]: Merge artefact

File: python/tskit/metadata.py
Comment (jeromekelleher) [inline python/tskit/metadata.py:166]: Merge artefact

**Review by jeromekelleher — COMMENTED** [review]
LGTM @ivan-krukov! I've gone through fairly carefully, and I don't see any show-stoppers.

There's some collatoral damage here from the merge where you've clobbered some recent metadata changes. You'll need to drop these from the diff (@benjeffery, any thoughts on the easiest way to do this?)

One thing we'll need to do as a follow up is to update simplify to keep track of the individual ID references in ``parents`` so that the IDs are correct afterwards.

Other than that, I think we just need to go through the standard steps for bumping file format version numbers and so on. These should all be done as follow ups, though, I think it's best to get these basic changes in soon so we don't have to keep rebasing them.

File: c/tests/testlib.c
Comment (ivan-krukov) [reply c/tests/testlib.c:93]: Changed accordingly

File: python/tskit/metadata.py
Comment (ivan-krukov) [reply python/tskit/metadata.py]: I've reverted the documentation filer, and both `metadata.py` and `test_metadata.py`. Should have been more careful about checking the patch. Thank you for taking a careful look into it.

File: c/tests/testlib.c
Comment (jeromekelleher) [reply c/tests/testlib.c:93]: Still says three?

**Review by jeromekelleher — APPROVED** [review]
LGTM @ivan-krukov! One tiny nit left I think, otherwise good to squash and merge. The squash should go cleanly, now.

@benjeffery, can you open up some issues to track the remaining things that need to be addressed before we release please?

**Review by benjeffery — APPROVED** [review]
Great stuff, I've been through closely and logged issues for follow up work.

**Comment by benjeffery** [comment]
As @jeromekelleher one nit about a comment that isn't consistent, then a squash and we can merge.
I recently added columns to the data model so I know how much work this is. Nice one!

**Comment by jeromekelleher** [comment]
@ivan-krukov, can we get this updated please? There's a significant chance of this PR getting out of date again if we don't merge it ASAP!

**Comment by benjeffery** [comment]
I've just force-pushed a fix and squash to this PR. Will merge if all green.

**Comment by jeromekelleher** [comment]
Segfault on some C tests - are you looking at this @benjeffery, or should @ivan-krukov pick it up?

**Comment by benjeffery** [comment]
I'll check it out.

**Comment by benjeffery** [comment]
The segfault was in the text parsing code for one of the example trees where the individuals string did not have parents (`test_simplest_unary_with_individuals`). I've fixed by adding `-1,-1` parents to that example.

**Comment by ivan-krukov** [comment]
I am so sorry that I am late for this. Thank you for resolving.

**Comment by jeromekelleher** [comment]
Woohoo, great to see this merged! Thanks for all the hard work @ivan-krukov, adding new columns is always a marathon. (Thanks too @benjeffery!)

**Comment by benjeffery** [comment]
> I am so sorry that I am late for this. Thank you for resolving.

No worries @ivan-krukov, it was a very small and quick fix. Thanks for the PR!

**Pull request merged**
